library/std/src/sys/unix/thread.rs

```diff
@@ -48,7 +55,12 @@ unsafe impl Sync for Thread {}
 
 impl Thread {
     // unsafe: see thread::Builder::spawn_unchecked for safety requirements
-    pub unsafe fn new(stack: usize, p: Box<dyn FnOnce()>) -> io::Result<Thread> {
+    pub unsafe fn new(
+        stack: usize,
+        p: Box<dyn FnOnce()>,
+        #[cfg(target_os = "horizon")] priority: i32,
+        #[cfg(target_os = "horizon")] affinity: i32,
```

Comment:
I thought about this, and that's how I expected it to work, but the issue is we need to now store more data in the builder. Since we're already directly modifying the builder, it made sense to keep the changes together here.

I could try a few things though to achieve an extension trait.

Reply:
I did some refactoring and got an extension trait to work.